src/3d/terrain/qgsflatterraingenerator.cpp

```diff
@@ -63,11 +63,23 @@ Qt3DCore::QEntity *FlatTerrainChunkLoader::createEntity( Qt3DCore::QEntity *pare
 
   // set up transform according to the extent covered by the quad geometry
   const QgsAABB bbox = mNode->bbox();
-  const double side = bbox.xMax - bbox.xMin;
-  const double half = side / 2;
 
-  transform->setScale( side );
-  transform->setTranslation( QVector3D( bbox.xMin + half, 0, bbox.zMin + half ) );
+  const QgsAABB mapFullExtent = Qgs3DUtils::mapToWorldExtent( map.extent(), bbox.yMin, bbox.yMax, map.origin() );
+
+  const QgsAABB commonExtent = QgsAABB( std::max( bbox.xMin, mapFullExtent.xMin ),
+                                        bbox.yMin,
+                                        std::max( bbox.zMin, mapFullExtent.zMin ),
+                                        std::min( bbox.xMax, mapFullExtent.xMax ),
+                                        bbox.yMax,
+                                        std::min( bbox.zMax, mapFullExtent.zMax )
+                                      );
+  const double xSide = commonExtent.xExtent();
+  const double zSide = commonExtent.zExtent();
+
+  transform->setScale3D( QVector3D( xSide, 1, zSide ) );
+  transform->setTranslation( QVector3D( commonExtent.xMin + xSide / 2, 0, commonExtent.zMin + zSide / 2 ) );
```

Comment:
why are you skipping the Y coordinate?

Reply:
The Y coordinate in 3d world is the elevation. In this case, the flat terrain chunked entity is a flat tile so we need to scale/translate its X and Z dimensions so that it takes a shape that may no longer be square (intersection between defined extent and quad geometry). No need to handle the Y elevation here.

Reply:
I missed the class name! Indeed for a flat tile it does not matter